lib/replace.ts

```diff
@@ -0,0 +1,104 @@
+import fs from "fs-extra";
+import { parse } from "@babel/parser";
+import traverse from "@babel/traverse";
+import * as t from "@babel/types";
+import { transformFromAst } from "@babel/core";
+
+async function replaceJSXTextInFile(
+  filePath: string,
+  replacement: { searchString: string; replaceWith: string }
+) {
+  const code = await fs.readFile(filePath, "utf-8");
+  const ast = parse(code, {
+    sourceType: "module",
+    plugins: ["jsx", "typescript"],
+  });
+
+  traverse(ast, {
+    JSXText(path) {
+      const { searchString, replaceWith } = replacement;
+
+      const regex = new RegExp(searchString, "gi");
+      if (regex.test(path.node.value)) {
+        const splitValues = splitByCaseInsensitive(
+          path.node.value,
+          searchString
+        );
+        const nodes: (t.JSXElement | t.JSXText)[] = [];
+
+        splitValues.forEach((splitValue) => {
+          if (splitValue.toLowerCase() === searchString.toLowerCase()) {
+            const identifier = t.jsxIdentifier("DittoComponent");
```

Comment:
Is it intentional that this only supports `ditto-react` components for now?

Reply:
This is intentional! I'll update to add support for i18next in a separate PR once I get things working end-to-end